recipes/remove-whitespace-from-string.md

```diff
@@ -0,0 +1,63 @@
+# Remove whitespace from string
+
+## Problem
+
+Remove extraneous whitespace (i.e. space, tab, newline) characters
+from a string.
+
+## Solution
+
+### Using SRFI 13
+
+These can remove arbitrary characters, but the default is to remove
+whitespace:
+
+* `(string-trim s)`
+* `(string-trim-right s)`
+* `(string-trim-both s)`
+
+### Using regular expressions
+
+```
+(define string-trim-both
+  (let ((r (regexp "^[ \t\r]*(.*?)[ \t\r]*$")))
+    (lambda (s)
+      (cadr (regexp-match r s)))))
+```
+
+The regular expression picks out the middle of the string. It is
+compiled only once, namely when `string-trim-both` is defined.
+
+Credit: [Jens Axel Søgaard](http://scheme.dk/)
```

Comment:
Credit: [Jens Axel Søgaard](http://racket-stories.com/)

Keep the credit line only if all recipes have credits - otherwise just delete the line.

Reply:
@jcubic added a credit line for each solution to each recipe. I think it's useful prudence. At the very least, it will enable to track down the author in case there are unresolved questions in the future.